tls/s2n_kex.c

```diff
@@ -41,23 +41,27 @@ static int s2n_check_ecdhe(const struct s2n_cipher_suite *cipher_suite, struct s
 
 static int s2n_check_kem(const struct s2n_cipher_suite *cipher_suite, struct s2n_connection *conn)
 {
-    notnull_check(conn);
-    /* There is no support for PQ KEMs while in FIPS mode */
-    if (s2n_is_in_fips_mode()) {
+    /* The int return value is treated like a bool; avoid using notnull_check(), etc */
```

Comment:
I was thinking something like that as well, because this is kind of ugly, but doing that would require a non-trivial refactor of the KEX code, and I didn't want to pollute this PR with those changes. I'll create an issue to follow up on this.

Reply:
:+1: for `s2n_result`!